src/lib/y2network/interfaces_collection.rb

```diff
@@ -87,5 +104,129 @@ def ==(other)
+
+    # Creates a map where the keys are the interfaces enslaved and the values
+    # are the bridges where them are taking part.
+    def bridge_index
+      index = {}
+
+      bridge_devs = Yast::NetworkInterfaces.FilterDevices("netcard").fetch("br", {})
+
+      bridge_devs.each do |bridge_master, value|
+        value["BRIDGE_PORTS"].to_s.split.each do |if_name|
+          index[if_name] = bridge_master
+        end
+      end
+
+      index
+    end
+
+    # Creates list of devices enslaved in the bond device.
+    #
+    # @param bond_iface [String] a name of an interface of bond type
+    # @return list of names of interfaces enslaved in the bond_iface
+    def bond_slaves(bond_iface)
+      bond_map = Yast::NetworkInterfaces::FilterDevices("netcard").fetch("bond", {}).fetch(bond_iface, {})
+
+      bond_map.select { |k, _| k.start_with?("BONDING_SLAVE") }.values
+    end
+
+    def bond_index
+      index = {}
+
+      bond_devs = Yast::NetworkInterfaces.FilterDevices("netcard").fetch("bond", {})
+
+      bond_devs.each do |bond_master, _value|
+        bond_slaves(bond_master).each do |slave|
+          index[slave] = bond_master
+        end
+      end
+
+      log.debug("bond slaves index: #{index}")
+
+      index
+    end
+
+    # Checks whether an interface can be bridged in particular bridge
+    #
+    # @param bridge_iface [Interface]
+    # @param iface [Interface] an interface to be validated as the bridge_iface slave
+    def bridgeable?(bridge_iface, iface)
+      return true if !iface.configured
+
+      if bond_index[iface.name]
+        log.debug("Excluding (#{iface.name}) - is bonded")
+        return false
+      end
+
+      # the iface is already in another bridge
+      if bridge_index[iface.name] && bridge_index[devname] != bridge_iface.name
+        log.debug("Excluding (#{iface.name}) - already bridged")
+        return false
+      end
+
+      # exclude interfaces of type unusable for bridge
+      case iface.type
+      when "br"
+        log.debug("Excluding (#{iface.name}) - is bridge")
+        return false
+      when "tun", "usb", "wlan"
+        log.debug("Excluding (#{iface.name}) - is #{iface.type}")
+        return false
+      end
+
+      case iface.startmode
+      when "nfsroot"
+        log.debug("Excluding (#{iface.name}) - is nfsroot")
+        return false
+
+      when "ifplugd"
+        log.debug("Excluding (#{iface.name}) - ifplugd")
+        return false
+      end
+
+      true
+    end
+
+    # Checks whether an interface can be enslaved in particular bond interface
+    #
+    # @param bond_iface [Interface]
+    # @param iface [Interface] an interface to be validated as bond_iface slave
+    # TODO: Check for valid configurations. E.g. bond device over vlan
+    # is nonsense and is not supported by netconfig.
+    # Also devices enslaved in a bridge should be excluded too.
+    def bondable?(bond_iface, iface)
+      Yast.import "Arch"
+      Yast.include self, "network/lan/s390.rb"
```

Comment:
this is not nice, as it basically modify this collection depending on method. So nasty side effect. If you need reuse existing include, I prefer to separate it to own object, so no accidental collision of instance variables - https://github.com/yast/yast-network/blob/network-ng/src/lib/y2network/widgets/slave_items.rb#L85